cdap-master/src/main/java/co/cask/cdap/data/tools/QueueConfigUpgrader.java

```diff
@@ -0,0 +1,133 @@
+                                                     TableId.from(datasetId.getNamespaceId(), dsName));
+    LOG.info("Starting upgrade for queue config table {}", Bytes.toString(queueConfigTable.getTableName()));
+    try {
+      Scan scan = new Scan();
+      scan.setTimeRange(0, HConstants.LATEST_TIMESTAMP);
+      scan.addFamily(QueueEntryRow.COLUMN_FAMILY);
+      scan.setMaxVersions(1); // we only need to see one version of each row
+      ResultScanner resultScanner = queueConfigTable.getScanner(scan);
+      List<Put> puts = Lists.newArrayList();
+      List<Delete> deletes = Lists.newArrayList();
+      Result result;
+      try {
+        while ((result = resultScanner.next()) != null) {
+          byte[] row = result.getRow();
+          String rowKey = Bytes.toString(row);
+          LOG.info("Processing queue config for  {}", rowKey);
+          NavigableMap<byte[], byte[]> columnsMap = result.getFamilyMap(QueueEntryRow.COLUMN_FAMILY);
+          QueueName queueName = fromRowKey(row);
+          if (queueName != null) {
+            Put put = new Put(queueName.toBytes());
+            LOG.info("Upgraded queue name: {}", queueName);
+            for (NavigableMap.Entry<byte[], byte[]> entry : columnsMap.entrySet()) {
+              LOG.info("Adding entry {} -> {} for upgrade",
+                       Bytes.toString(entry.getKey()), Bytes.toString(entry.getValue()));
+              put.add(QueueEntryRow.COLUMN_FAMILY, entry.getKey(), entry.getValue());
+              puts.add(put);
+            }
+            LOG.info("Marking old key {} for deletion", rowKey);
+            deletes.add(new Delete(row));
+          }
+          LOG.info("Finished processing queue config for {}", rowKey);
+        }
+      } finally {
+        resultScanner.close();
+      }
+
+      queueConfigTable.put(puts);
+      queueConfigTable.delete(deletes);
+
+      LOG.info("Successfully completed upgrade for queue config table {}",
+               Bytes.toString(queueConfigTable.getTableName()));
+    } catch (Exception e) {
+      LOG.error("Error while upgrading queue config table: {}", datasetId, e);
+      throw Throwables.propagate(e);
+    } finally {
+      queueConfigTable.close();
+    }
+  }
+
+  @Nullable
+  private QueueName fromRowKey(byte[] oldRowKey) {
+    QueueName queueName = QueueName.from(oldRowKey);
+
+    if (queueName.getNumComponents() == 4) {
+      LOG.info("Found old queue config {}. Upgrading.", queueName.toString());
+      return QueueName.fromFlowlet(Constants.DEFAULT_NAMESPACE,
+                                   queueName.getFirstComponent(),
+                                   queueName.getSecondComponent(),
+                                   queueName.getThirdComponent(),
+                                   queueName.getFourthComponent());
+    } else if (queueName.getNumComponents() == 5) {
+      LOG.info("Queue config for {} seems to already have been upgraded. Skipping.", queueName.toString());
```

Comment:
When logging with `{}`, no need to have `.toString`. Actually it's better not to have `.toString()` so that it only get invoked when the log level requires the log output.